src/utility/firebase.js

```diff
@@ -92,3 +97,51 @@ export const getProject = async (user_id, setProjectCallback, setFeedbackCallbac
       })
   }
 }
+
+const createNewApplication = async user => {
+  const userId = {
+    _id: user.uid,
+  }
+  const basicInfo = {
+    basicInfo: {
+      email: user.email,
+      firstName: user.displayName.split(' ')[0],
+      lastName: user.displayName.split(' ')[1],
+    },
+  }
+  const submission = {
+    submission: {
+      lastUpdated: firebase.firestore.Timestamp.now(),
+      submitted: false,
+    },
+  }
+
+  const newApplication = {
+    ...hackerApplicationTemplate,
+    ...basicInfo,
+    ...submission,
+    ...userId,
+  }
+
+  await applicantsRef.doc(user.uid).set(newApplication)
+}
+
+export const getUserStatus = async user => {
+  const applicant = await applicantsRef.doc(user.uid).get()
+  if (!applicant.exists) {
+    await createNewApplication(user)
+    return applicantStatus.new
+  }
+  if (applicant.data().status.attending) {
+    return applicantStatus.attending
+  }
+
+  const status = applicant.data().status.applicationStatus
+  if (status === applicantStatus.applied) {
+    return applicantStatus.applied
+  }
+  if (status === applicantStatus.accepted) {
+    return applicantStatus.accepted
+  }
+  return applicantStatus.inProgress
+}
```

Comment:
do you think we can just do something like this, using `applicationStatus` from Constants.js?
```
return applicationStatus[status]
```

Reply:
I don't want to have to support every status because we don't need to, so this function essentially muxes them together